### DISCUSSION TITLE
Is Ctrl-C for loops in the REPL in xonsh v0.16?

### DISCUSSION THREAD

**Discussion by halloleo** [xonsh/xonsh#5511] [Q&A] [answered]
Hi! I was super impressed when I saw the fix from #5380. This is extremely admirable work. Thank you so much!

Do I understand it correctly this fixed the "behaviour" that loops in the REPL cannot be interrupted via Ctrl-C, right?

If so, is it meant to be in the newest release v0.16?

I couldn't make it work (neither with or without `$RAISE_SUBPROC_ERROR=True`).

**Comment by anki-code** [comment, marked as answer]
Hey! The fix will be in next release (0.17.0). To test this now you can install xonsh from main branch i.e.:
```xsh
xpip install -U --force-reinstall git+https://github.com/xonsh/xonsh
# restart xonsh
```

In the nutshell the fix from #5380 allows to return the right exit code for subprocess call and if you set `$RAISE_SUBPROC_ERROR=True` before loop and exit code is not 0 for command in the loop the loop will be interrupted.

**Reply by halloleo** [reply]
Thanks for the details - and the clarification on `$RAISE_SUBPROC_ERROR`. 

(Upgrading xonsh via its own xpip looks cool, but my xonsh is installed via Homebrew and `brew` doesn't seem to allow this reinstall. - No drama, I'll wait for the release.)

**Reply by anki-code** [reply]
@halloleo As I see in https://repology.org/project/xonsh/versions brew should have the latest version.

**Reply by halloleo** [reply]
True! But that must have happened in the very last few days! Thanks for pointing it out.
